src/features/mcda/atoms/mcdaCalculation/mapStyle.ts

```diff
@@ -34,8 +34,79 @@ export function linearNormalization(layers: MCDAConfig['layers']) {
   }
 }
 
+type PaintProps = {
+  colorsConfig: MCDAConfig['colors'];
+  mcdaResult: number | (string | number | (string | number)[])[];
+  absoluteMax: number;
+  absoluteMin: number;
+};
+
+function sentimentPaint({
+  colorsConfig,
+  mcdaResult,
+  absoluteMin,
+  absoluteMax,
+}: PaintProps) {
+  const { good = DEFAULT_GREEN, bad = DEFAULT_RED } = colorsConfig.parameters;
+  return {
+    'fill-color': [
+      'let',
+      'mcdaResult',
+      ['to-number', mcdaResult, -9999], // falsy values become -9999
+      [
+        'case',
+        [
+          'all',
+          ['>=', ['var', 'mcdaResult'], absoluteMin],
+          ['<=', ['var', 'mcdaResult'], absoluteMax],
+        ],
+        ['interpolate-hcl', ['linear'], ['var', 'mcdaResult'], 0, bad, 1, good],
+        'transparent', // all values outside of range [0,1] will be painted as transparent
+      ],
+    ],
+    'fill-opacity': 1,
+    'fill-antialias': false,
+  };
+}
+
+function expressionsPaint({
+  colorsConfig,
+  mcdaResult,
+  absoluteMax,
+  absoluteMin,
+}: PaintProps) {
+  return Object.entries(colorsConfig.parameters).reduce(
+    (acc, [paintProp, expression]) => {
+      acc[paintProp] = Array.isArray(expression)
+        ? [
+            'let',
+            'mcdaResult',
+            ['to-number', mcdaResult, -9999], // falsy values become -9999,
+            'absoluteMax',
+            ['to-number', absoluteMax, -9999], // falsy values become -9999,
+            'absoluteMin',
+            ['to-number', absoluteMin, -9999], // falsy values become -9999,
+            ...expression,
+          ]
+        : expression;
+
+      return acc;
+    },
+    {} as Record<any, any>,
+  );
+}
+
+function generateLayerPaint(props: PaintProps) {
```

Comment:
No default case for switch - is it safe here?

Reply:
Since the data is validated on the input against the types, it is safe. When we add new invariant in types - ts rise error here